src/core/opamSystem.ml

```diff
@@ -488,9 +488,12 @@ let t_resolve_command =
         name ^ ".exe"
       else name
     in
-    let possibles = OpamStd.List.filter_map (fun path ->
-        let candidate = Filename.concat path name in
-        if Sys.file_exists candidate then Some candidate else None) path
+    let possibles =
+      OpamStd.List.filter_map (fun path ->
+          let candidate = Filename.concat path name in
+          if Sys.file_exists candidate && not (Sys.is_directory candidate) then
+            Some candidate else None)
```

Comment:
This is slightly redundant, as `Sys.is_directory` will fail if the file doesn't exist
```suggestion
          match Sys.is_directory candidate with
          | false -> Some candidate
          | true -> None
          | exception (Sys_error _) -> None)
```

Reply:
A more compressed version can be used instead:
```suggestion
          match Sys.is_directory candidate with
          | false -> Some candidate
          | true | exception (Sys_error _) -> None)
```